src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -278,14 +278,17 @@ public void WriteBindings(IResource resource)
                 Writer.WriteString("protocol", endpoint.Protocol.ToString().ToLowerInvariant());
                 Writer.WriteString("transport", endpoint.Transport);
 
-                int? targetPort = (resource, endpoint.UriScheme, endpoint.TargetPort) switch
+                int? targetPort = (resource, endpoint.UriScheme, endpoint.TargetPort, endpoint.Port) switch
                 {
                     // The port was specified so use it
-                    (_, _, int port) => port,
+                    (_, _, int target, _) => target,
 
-                    // Project resources get their default port from the deployment tool
+                    // Container resources get their default listening port from the exposed port.
+                    (ContainerResource, _, _, int port) => port,
```

Comment:
Hmm not sure this is right. We care about the target port itself being null in this case.

Reply:
Doesn't it have to be `null` in this case? If it wasn't 284 would have taken it.

Reply:
Yes, it would already have matched, but it's good to be explicit anyway.